File: bubble_sort/python/bubble_sort.py
def bubble_sort(input_list):
    """
    Sorts a list of elements in ascending order using the bubble sort algorithm.

    This algorithm works by repeatedly swapping adjacent elements if they are in the wrong order.

    Args:
        input_list (list): The list of elements to be sorted.

    Returns:
        list: A sorted list of elements in ascending order.
    """
    n = len(input_list)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if input_list[j] > input_list[j + 1]:
                input_list[j], input_list[j + 1] = input_list[j + 1], input_list[j]
            print('the state of the array-->', input_list)
    return input_list

not_sort_array = [5, 3, 8, 4, 2]

print('not_sort_array-->', not_sort_array)
print('sort_array-->', bubble_sort(not_sort_array))

File: bubble_sort/javascript/bubble_sort.js
const nonSortedArray = [5, 3, 8, 4, 2]
const inputList = [...nonSortedArray]

/**
 * Sorts a given array using bubble sort algorithm.
 * @param {Array} inputList array of numbers to be sorted
 * @return {Array} sorted array
 */
function bubbleSort(sortedList) {
  for (let i = 0; i < (sortedList.length - 1); i += 1) {
    console.log(`Iteration ${i + 1}:`);
    for (let j = 0; j < (sortedList.length - i - 1); j += 1) {
      console.log(`  Comparing ${sortedList[j]} and ${sortedList[j + 1]}`);
      if (sortedList[j] > sortedList[j + 1]) {
        console.log(`  Swapping ${sortedList[j]} and ${sortedList[j + 1]}`);
        const temp = sortedList[j];
        sortedList[j] = sortedList[j + 1];
        sortedList[j + 1] = temp;
      }
    }
  }
  return sortedList;
}

console.log('not sorted array: ', nonSortedArray)
console.log('sorted array: ', bubbleSort(inputList))


const outputElement = document.getElementById('output');
outputElement.innerHTML = '';
outputElement.innerHTML = `
  <p>Not sorted array: ${nonSortedArray}</p>
  <p>Sorted array: ${inputList}</p>
  <p>For details, check the console.</p>
`;